### PR TITLE
Fix #2704: ensure route registered via $routes->cli() not accessible via web browser even autoroute is true

### LINKED ISSUE #2704: Bug: $routes->cli() accessible via web browser if autoroute is true
For example, I have the following  `cli` route: 

```php 
// set auto route to true
$routes->setAutoRoute(true); 

// set "hello" cli route
$routes->cli('hello', 'Commands\Hello::index');
```

and I have the controller: 

```php 
<?php
namespace App\Controllers\Commands;

use App\Controllers\BaseController;

class Hello extends BaseController
{
    public function index()
    {
        echo 'Hello';
    }
}
```

then, I open it via web browser, it still accesible via : `http://ci4.local/commands/hello` , it still accessible.

**CodeIgniter 4 version**
4.0.2

**Expected behavior, and steps to reproduce if appropriate**

By define `cli()`, it should decline the http access into it. The documentation doesn't say it require autoroute to be false: 

```
You can create routes that work only from the command-line, and are inaccessible from the web browser, with the cli() method
```

**Context**
 - OS: OSX
 - Web server Apache
 - PHP version 7.4

### PR DESCRIPTION
**Pull request by samsonasik**
Fix #2704: ensure route registered via $routes->cli() not accessible via web browser even autoroute is true.

**Checklist:**
- [x] Securely signed commits
- [x] Unit testing, with >80% coverage

### CODE REVIEW

**Comment by samsonasik** [comment]
**Before** : 
Screen Shot 2020-03-15 at 5 58 15 PM

**After**: 
Screen Shot 2020-03-15 at 5 58 25 PM

**Comment by samsonasik** [comment]
Not sure how to write unit test for it. I included screenshot before and after changed above.

**Comment by samsonasik** [comment]
@MGatner I've added feature test case for demoing 404 page thrown when open cli routes from http

**Comment by samsonasik** [comment]
it seems need some autoload tweak for testing "App" namespace for feature test demo, I will re-check again.

**Comment by samsonasik** [comment]
The test tweak implemented.

**Comment by samsonasik** [comment]
@@MGatner travis build green 🎉

File: system/Router/Router.php [hunk 1]
```diff
@@ -542,6 +542,25 @@ public function autoRoute(string $uri)
...
+			$controller  = '\\' . $this->collection->getDefaultNamespace();
+			$controller .= $this->directory ? str_replace('/', '\\', $this->directory) : '';
+			$controller .= $this->controllerName();
+			$methodName  = $this->methodName();
+
+			foreach ($this->collection->getRoutes('cli') as $route)
+			{
+				if (is_string($route))
+				{
+					if (strpos($route, $controller . '::' . $methodName) === 0)
```
Comment (samsonasik) [inline system/Router/Router.php]: using `strpos` is on purpose to handle parameter, eg: controller::method/$1: 

```php
$routes->cli('hello/(:any)', 'Commands\Hello::index/$1');
```

**Comment by samsonasik** [comment]
I've added feature test case for demoing 404 page thrown when open parameterized cli routes from http

**Comment by samsonasik** [comment]
I've added condition check for default method index when routing defined in cli as controller name only, which fallback to method index, eg:

```php 
$routes->cli('hello', 'Hello');
```

**Comment by samsonasik** [comment]
I've added condition for multi case, eg: capitalized controller or method in URI with compare with strtolower of the controller and method.

**Comment by samsonasik** [comment]
travis build green 🎉

File: tests/system/Test/FeatureTestCaseTest.php [hunk 2]
```diff
@@ -204,4 +204,54 @@ public function testCallZeroAsPathGot404()
...
+			],
+		];
+	}
+
+	/**
+	 * @dataProvider provideRoutesData
+	 */
+	public function testOpenCliRoutesFromHttpGot404($from, $to, $httpGet)
+	{
+		$this->expectException(PageNotFoundException::class);
```
Comment (kenjis) [inline tests/system/Test/FeatureTestCaseTest.php:239]: These tests pass because there is no `App\Controllers\Hello` class.

Comment (kenjis) [inline tests/system/Test/FeatureTestCaseTest.php:239]: These tests pass because there is no `App\Controllers\Hello` class.

**Pull request merged**